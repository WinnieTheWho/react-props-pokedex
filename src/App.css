h1 {
  text-align: center;
}

.pokedex {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

.card {
  background-color: lightgray;
  border-radius: 25px;
  text-align: center;
  min-width: 200px;
  margin: 10px;
}
