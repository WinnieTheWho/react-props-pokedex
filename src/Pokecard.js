import React from "react";

class Pokecard extends React.Component {
  render() {
    const { name, type, base_experience, img } = this.props;

    return (
      <div className="card">
        <h3>{name}</h3>
        <img src={img} alt="pokemon"></img>
        <p>Type: {type}</p>
        <p>Experience: {base_experience}</p>
      </div>
    );
  }
}

export default Pokecard;
