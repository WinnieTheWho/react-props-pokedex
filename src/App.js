import React from 'react';
import './App.css';
import Pokedex from './Pokedex';

function App() {
  return (
    <div>
      <h1> POKEDEX </h1>
      <div>
          <Pokedex />
      </div>
    </div>
  );
}

export default App;
