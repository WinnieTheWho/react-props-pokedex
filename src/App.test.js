import React from 'react';
import { shallow, mount } from "enzyme"
import App from './App';
import toJson from "enzyme-to-json";
import Pokecard from './Pokecard';

//smoke test shallow 
it('renders without crashing', () => {
  shallow (<App />);
});

//smoke test mount 
it('renders without crashing', () => {
  mount (<App />);
});

//snapshot test 
it("matches snapshot", function() {
  let wrapper = shallow(   <Pokecard
    name="Pikachu"
    img="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
    type="electric"
    base_experience="112"
  />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

//snapshot test 
it("matches snapshot", function() {
  let wrapper = shallow(   <Pokecard
    name="Charmander"
    img="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
    type="fire"
    base_experience="62"
  />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
