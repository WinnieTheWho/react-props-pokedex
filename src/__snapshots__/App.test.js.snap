// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card"
>
  <h3>
    Pikachu
  </h3>
  <img
    alt="pokemon"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
  />
  <p>
    Type: 
    electric
  </p>
  <p>
    Experience: 
    112
  </p>
</div>
`;

exports[`matches snapshot 2`] = `
<div
  className="card"
>
  <h3>
    Charmander
  </h3>
  <img
    alt="pokemon"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
  />
  <p>
    Type: 
    fire
  </p>
  <p>
    Experience: 
    62
  </p>
</div>
`;
